### PR TITLE
[Relay][Pass]Improve memory_allocation pass to support multiple i/o dynamic kernels

### PR DESCRIPTION
**Pull request by kevinthesun**
This PR enhance memory_allocation pass to support dynamic shape kernels such as split and concatenate.

@zhiics @jroesch @yongwww

### CODE REVIEW

File: python/tvm/relay/memory_alloc.py [hunk 1]
```diff
@@ -220,7 +231,11 @@ def visit_call(self, call):
                 scope.let("shape_func", shape_call)
 
                 out_types = []
-                out_types.append(call.checked_type)
+                if isinstance(ret_type, ty.TensorType):
```
Comment (zhiics) [inline python/tvm/relay/memory_alloc.py]: Can we use `LinearizeRetType` here?

File: python/tvm/relay/op/_tensor.py [hunk 2]
```diff
@@ -196,6 +200,9 @@ def elemwise_shape_func(attrs, inputs, _):
 register_shape_func("less_equal", False, broadcast_shape_func)
 register_shape_func("greater", False, broadcast_shape_func)
 register_shape_func("greater_equal", False, broadcast_shape_func)
+register_shape_func("maximum", False, broadcast_shape_func)
```
Comment (yongwww) [inline python/tvm/relay/op/_tensor.py]: add test case for maximum/mininum/exp

Comment (kevinthesun) [reply python/tvm/relay/op/_tensor.py]: broadcast has already been covered. Currently in test_any we don't test every op in the same category. I add test any for elemwise

**Comment by zhiics** [comment]
Thanks @kevinthesun @yongwww
